mailpoet/lib/Config/SubscriberChangesNotifier.php

```diff
@@ -0,0 +1,113 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Config;
+
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Carbon\Carbon;
+
+class SubscriberChangesNotifier {
+
+  /** @var array<int, int> */
+  private $createdSubscriberIds = [];
+
+  /** @var array<int, int> */
+  private $deletedSubscriberIds = [];
+
+  /** @var array<int, int> */
+  private $updatedSubscriberIds = [];
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    WPFunctions $wp
+  ) {
+    $this->wp = $wp;
+  }
+
+  public function notify() {
+    $this->notifyCreations();
+    $this->notifyUpdates();
+    $this->notifyDeletes();
+  }
+
+  private function notifyCreations(): void {
+    if (count($this->createdSubscriberIds) === 1) {
+      foreach ($this->createdSubscriberIds as $subscriberId => $updatedAt) {
+        $this->wp->doAction(SubscriberEntity::HOOK_SUBSCRIBER_CREATED, $subscriberId);
+      }
+    } elseif ($this->createdSubscriberIds) {
+      $minTimestamp = min($this->createdSubscriberIds);
+      if ($minTimestamp) {
+        $this->wp->doAction(SubscriberEntity::HOOK_MULTIPLE_SUBSCRIBERS_CREATED, $minTimestamp);
+      }
+    }
+  }
+
+  private function notifyUpdates(): void {
+    // unset updated subscribers if subscriber is created
+    foreach ($this->createdSubscriberIds as $subscriberId => $timestamp) {
+      unset($this->updatedSubscriberIds[$subscriberId]);
+    }
+
+    if (count($this->updatedSubscriberIds) === 1) {
+      foreach ($this->updatedSubscriberIds as $subscriberId => $updatedAt) {
+        $this->wp->doAction(SubscriberEntity::HOOK_SUBSCRIBER_UPDATED, $subscriberId);
+      }
+    } elseif ($this->updatedSubscriberIds) {
+      $minTimestamp = min($this->updatedSubscriberIds);
+      if ($minTimestamp) {
+        $this->wp->doAction(SubscriberEntity::HOOK_MULTIPLE_SUBSCRIBERS_UPDATED, $minTimestamp);
+      }
+    }
+  }
+
+  private function notifyDeletes(): void {
+    if (count($this->deletedSubscriberIds) === 1) {
+      foreach ($this->deletedSubscriberIds as $subscriberId => $updatedAt) {
+        $this->wp->doAction(SubscriberEntity::HOOK_SUBSCRIBER_DELETED, $subscriberId);
+      }
+    } elseif ($this->deletedSubscriberIds) {
+      $this->wp->doAction(SubscriberEntity::HOOK_MULTIPLE_SUBSCRIBERS_DELETED, array_keys($this->deletedSubscriberIds));
```

Comment:
if there are no deleted subscribers... we will get the hook `HOOK_MULTIPLE_SUBSCRIBERS_DELETED` after calling to the `notify()` method, won't we?